service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceClient.java

```diff
@@ -369,7 +369,10 @@ public void send(String deviceId, String moduleId, Message message) throws IOExc
     public CompletableFuture<Void> openAsync()
     {
         final CompletableFuture<Void> future = new CompletableFuture<>();
-        executor.submit(() -> {
+
+        // executor service hasn't been initialized yet, so don't use it to run this thread
```

Comment:
when is the executor service initialized? 
does this thread now run unmonitored? what will happen if this gets interrupted?

Reply:
The executor service is initialized in the ```open()``` (synchronous) call. It can't be interrupted because the open call doesn't throw an interrupted exception, and we don't expose this thread to the user to allow them to interrupt it either. The open() call that is run asynchronously actually has no IO operations in it. It basically just initializes the executor service.